sstables/sstables.cc

```diff
@@ -3637,6 +3638,18 @@ generation_type::from_string(const std::string& s) {
     }
 }
 
+future<data_sink> file_io_extension::wrap_sink(const sstable& sst, component_type c, data_sink sink) {
+    file dummy = create_noop_file();
+    auto f = co_await wrap_file(sst, c, std::move(dummy), open_flags::wo);
+
+    if (!f) {
+        co_return sink;
+    }
+    co_await f.close();
```

Comment:
This "detect if we wrap files for this component" is pretty horrible. Why can't `file_io_extension::wrap_sink` be pure  virtual just like its `wrap_file` is?

Reply:
Because there are a bunch of (a few at least) extensions used in testing (to fake errors mainly) which, if no default implementation is provided, would have to implement the method, and probably use the same, not great wrapper. The default is not good for efficiency, but it works fine for those tests etc. And then this change does not break any existing code.

Reply:
> Because there are a bunch of (a few at least) extensions used in testing ...

Ah, OK